http/get_compressed/python/client/client.py

```diff
@@ -0,0 +1,96 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import urllib.request
+import pyarrow as pa
+import time
+
+URI = "http://localhost:8008"
+ARROW_STREAM_FORMAT = "application/vnd.apache.arrow.stream"
+
+
+def make_request(uri, compression):
+    coding = "identity" if compression.startswith("identity") else compression
+    # urllib.request.urlopen() always sends an HTTP/1.1 request
+    # with Accept-Encoding: identity, so we need to setup a request
+    # object with custom headers to request a specific compression
+    headers = {
+        "Accept-Encoding": f"{coding}, *;q=0",
+    }
+    if compression.startswith("identity+"):
+        # request IPC buffer compression instead of HTTP compression
+        ipc_codec = compression.split("+")[1]
+        headers["Accept"] = f'{ARROW_STREAM_FORMAT};codec="{ipc_codec}"'
```

Comment:
Is it `codec` or `codecs`?

Reply:
Singular because it's in the response, but I could rename it to `codecs` to simplify things.

Reply:
I renamed it to codecs. Technically, each buffer could have their own compression algorithm, so plural still makes sense on the response.